great_tables/_utils_render_latex.py

```diff
@@ -0,0 +1,1001 @@
+from __future__ import annotations
+
+from itertools import chain
+
+import re
+
+from ._gt_data import GTData, GroupRowInfo
+from ._tbl_data import _get_cell, cast_frame_to_string, replace_null_frame
+from .quarto import check_quarto
+from great_tables._spanners import spanners_print_matrix
+from great_tables._utils import heading_has_subtitle, heading_has_title, seq_groups
+from great_tables._utils_render_html import _get_spanners_matrix_height
+from great_tables._text import _process_text
+
+from typing import TypedDict, List
+
+
+LENGTH_TRANSLATIONS_TO_PX = {
+    "px": 1.0,
+    "pt": 4 / 3,
+    "in": 96.0,
+    "cm": 37.7952755906,
+    "emu": 1 / 9525,
+    "em": 16.0,
+}
+
+
+class WidthDict(TypedDict):
```

Comment:
Can you change this away from

* WidthDict: A class full of lists that will all become `len(range(n))`

To a list of width objects:

* Use a class of non-list entries. E.g. a dataclass called EntryWidth, with type, unspec, etc..
* A list of len n, where each entry is an EntryWidth instance
* Totally legit to define a container dataclass like XxWidths, that
  - holds the list[EntryWidth] and tbl_width
  - name w/e feels more appropriate

Reply:
* The type hint for `type:` should be `ColInfoTypeEnum`
* The type hint for `column_align` should be _gt_data.ColumnAlignment

Reply:
It looks like pieces like `type:` are pulled out specifically, so we later can essentially manually do a `boxhead[i].visible` check. I wonder if just storing the original ColInfo object for each element would let us do these checks directly (and then we don't need to store just the `type` piece).

Reply:
From some refactoring, this whole piece was removed (it's never actually used in the code). It will be used once we have the stub and row groups rendered, but right now it's unused code.